File: leetcode/src/com/leetcode/question/LeetCode9.java
package com.leetcode.question;

import java.util.Scanner;

/**
 * Palindrome Number
 */
public class LeetCode9 {
    public static void main(String... args){
        Scanner scanner = new Scanner(System.in);

        while (true){
            System.out.println("===============================================");
            System.out.println("请输入数字：");
            int num = scanner.nextInt();

            if (num == 0){
                break;
            }
            System.out.println("翻转后的数值为：" + isPalindrome(num));
        }
        scanner.close();
    }

    private static boolean isPalindrome(int x){

        return true;
    }

}

File: leetcode_kotlin/src/com/example/leetcode/Leetcode9.kt
package com.example.leetcode

import java.util.*

fun main(args:Array<String>){
    val scanner:Scanner = Scanner(System.`in`);
    while (true){
        println("请输入判断数字");
        val a = scanner.nextInt();
        if (a == 0){
            break;
        }
        println("回文数返回结果：" + isPalindrome(a));
    }
    scanner.close();
}

fun isPalindrome(x:Int):Boolean{
    if (x < 0){
        //负数不是回文数，例如-121，是121-
        return false;
    }
    var x1 = x;
    var target:Int = 0;
    while (x1/10 != 0){
        target = target*10+x1%10;
        x1 = x1/10;
    }
    target = target*10+x1;
    print(target)
    if (x == target){
        return true;
    }
    return false;
}